protocols/gossipsub/src/behaviour.rs

```diff
@@ -3126,12 +3199,94 @@ where
     }
 }
 
+/// This is called when peers are added to any mesh. It checks if the peer existed
+/// in any other mesh. If this is the first mesh they have joined, it queues a message to notify
+/// the appropriate connection handler to maintain a connection.
+fn peer_added_to_mesh(
+    peer_id: PeerId,
+    new_topics: Vec<&TopicHash>,
+    mesh: &HashMap<TopicHash, BTreeSet<PeerId>>,
+    known_topics: Option<&BTreeSet<TopicHash>>,
+    events: &mut VecDeque<GossipsubNetworkBehaviourAction>,
+    connections: &HashMap<PeerId, PeerConnections>,
+) {
+    // Ensure there is an active connection
+    let connection_id = {
+        let conn = connections.get(&peer_id).expect("To be connected to peer.");
+        assert!(
+            !conn.connections.is_empty(),
+            "Must have at least one connection"
+        );
+        conn.connections[0]
+    };
```

Comment:
```suggestion
    let connection_id  = connections.get(&peer_id)
        .expect("To be connected to peer.")
        .connections.get(0)
        .expect("Must have at least one connection");
```

Would that not be easier to read?